src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndex.java

```diff
@@ -38,7 +70,51 @@ public abstract void saveIndex(int[] ids, float[][] data, String indexPath, Stri
      *
      * @return size of the hnsw index on the disk in KB.
      */
-    public abstract long getIndexSize();
+    public long getIndexSize() {
+        return this.indexSize;
+    }
+
+    public SpaceType getSpaceType() {
+        return this.spaceType;
+    }
+
+    public void close() {
+        Lock writeLock = readWriteLock.writeLock();
+        writeLock.lock();
+        // Autocloseable documentation recommends making close idempotent. We don't expect to doubly close
+        // but this will help prevent a crash in that situation.
+        if (this.isClosed) {
+            return;
+        }
+        try {
+            gcJniWrapper(this.indexPointer);
+        } finally {
+            this.isClosed = true;
+            writeLock.unlock();
+        }
+    }
+
+    /**
+     * determines the size of the engine index on disk
+     * @param indexPath absolute path of the index
+     *
+     */
+    protected static long computeFileSize(String indexPath) {
```

Comment:
This looks like a utility method. we can keep this class have only required methods, what do you think?

Reply:
You are right. This can be moved out of this class to reduce dependencies. I will move it out.